homeassistant/components/media_player/firetv.py

```diff
@@ -0,0 +1,214 @@
+    b. Select System > About > Network.
+3. `pip install firetv[firetv-server]` into a Python 2.x environment
+4. `firetv-server -d <fire tv device IP>:5555`, background the process
+5. Configure Home Assistant as follows:
+
+media_player:
+  platform: firetv
+  # optional: where firetv-server is running (default is 'localhost:5556')
+  host: localhost:5556
+  # optional: device id (default is 'default')
+  device: livingroom-firetv
+  # optional: friendly name (default is 'Amazon Fire TV')
+  name: My Amazon Fire TV
+
+Note that python-firetv has support for multiple Amazon Fire TV devices.
+If you have more than one configured, be sure to specify the device id used.
+Run `firetv-server -h` and/or view the source for complete capabilities.
+
+Possible states are:
+ - off (TV screen is dark)
+ - standby (standard UI is active - not apps)
+ - idle (screen saver is active)
+ - play (video is playing)
+ - pause (video is paused)
+ - disconnected (can't communicate with device)
+"""
+
+import requests
+
+from homeassistant.const import (
+    STATE_PLAYING, STATE_PAUSED, STATE_IDLE, STATE_OFF,
+    STATE_UNKNOWN, STATE_STANDBY)
+
+from homeassistant.components.media_player import (
+    MediaPlayerDevice,
+    SUPPORT_PAUSE, SUPPORT_VOLUME_SET,
+    SUPPORT_TURN_ON, SUPPORT_TURN_OFF,
+    SUPPORT_PREVIOUS_TRACK, SUPPORT_NEXT_TRACK)
+
+SUPPORT_FIRETV = SUPPORT_PAUSE | \
+    SUPPORT_TURN_ON | SUPPORT_TURN_OFF | SUPPORT_PREVIOUS_TRACK | \
+    SUPPORT_NEXT_TRACK | SUPPORT_VOLUME_SET
+
+DOMAIN = 'firetv'
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the firetv platform. """
+    add_devices([
+        FireTVDevice(
+            config.get('host', 'localhost:5556'),
+            config.get('device', 'default'),
+            config.get('name', 'Amazon Fire TV')
+        )
+    ])
+
+
+class FireTV(object):
+    """ firetv-server client.
+
+    Should a native Python 3 ADB module become available,
+    python-firetv can support Python 3, it can be added
+    as a dependency, and this class can be dispensed of.
+
+    For now, it acts as a client to the firetv-server
+    HTTP server (which must be running via Python 2).
+    """
+
+    DEVICE_STATE_URL = 'http://{0}/devices/state/{1}'
+    DEVICE_ACTION_URL = 'http://{0}/devices/action/{1}/{2}'
+
+    def __init__(self, host, device_id):
+        self.host = host
+        self.device_id = device_id
+
+    @property
+    def state(self):
+        """ Get the device state.
+
+        An exception means UNKNOWN state.
+        """
+        try:
+            response = requests.get(
+                FireTV.DEVICE_STATE_URL.format(
+                    self.host,
+                    self.device_id
+                    )
+                )
+            return response.json()['state']
+        except requests.exceptions.HTTPError:
+            return STATE_UNKNOWN
+        except requests.exceptions.RequestException:
```

Comment:
This except includes the previous exception. Also, I think it would be nice to log an error that we're unable to communicate with firetv-server so users can easier debug their setup.